lib/chef/resource/aws_ebs_volume.rb

```diff
@@ -17,9 +19,11 @@ class Chef::Resource::AwsEbsVolume < Chef::Provisioning::AWSDriver::AWSResourceW
     name =~ /^vol-[a-f0-9]{8}$/ ? name : nil
   }
 
+  attribute :device,            kind_of: String
+
   def aws_object
     driver, id = get_driver_and_id
     result = driver.ec2.volumes[id] if id
-    result && result.exists? && ![:destroyd, :deleting].include?(result.status) ? result : nil
+    result && result.exists? && ![:deleted, :deleting].include?(result.status) ? result : nil
```

Comment:
If an ebs volume has a status of `:deleted` or `:deleting`, will it eventually remove itself completely from AWS?  Thats how instances seem to work.  If volumes work the same way this seems good.